### ISSUE TITLE
Install location requires sudo

### ISSUE THREAD

**Issue by spinicist** [vector-of-bool/vscode-cmake-tools#352]
### Brief Issue Summary

I would like to install my project to `/usr/local` using the `install` CMake target. On my (Mac) system this requires `sudo` permissions. Instead of asking for permissions, CMake Tools simply fails. I can run `sudo ninja install` from the command-line. However, if CMake has to build a binary during this step, it will be written with the owner set to `root`. Then, if I make another target within CMake Tools, e.g. `all`, it will fail because it cannot overwrite the file.

### Expected:

Somehow CMake Tools asks for my password so that I can install to `/usr/local`.

### Other Notes/Information

I admit this may be a difficult problem to solve well. The reason I would like to install to `/usr/local` is that I use BATS as my test framework, and since Apple added System Integrity Protection (SIP) to Mac OS X, child processes no longer receive the `DYLD_LIBRARY_PATH` variable in their environment. This means my BATS tests cannot find my shared libraries, unless they are installed to a system path such as `/usr/local` 😡

If anyone has a better suggestion as a workaround, I am all ears.

**Comment by ytimenkov** [comment]
Recently code introduced ability to save files as administrator. Wonder if could implement similar feature on all platforms.
Only issue though is to detect when it is needed...

**Comment by spinicist** [comment]
Yup, that's exactly the problem I thought of once I'd had a while to think. Do not consider this high priority! I have found another work-around (re-sourcing `.bash_profile` in my BATS tests).

**Comment by Tudyx** [comment]
Any update on this feature request?

**Comment by andreeis** [comment]
We were not able to get to this yet but we would happily review and test a PR from the community, if anyone is willing to spend the time to implement this.

**Comment by gvcallen** [comment]
Any updates on this feature? I almost always install to a location that requires sudo privileges

**Comment by DominionSoftware** [comment]
would like to see this also.

**Comment by Zingam** [comment]
I imagine that this feature request is somewhat related to: https://github.com/microsoft/vscode-cmake-tools/issues/532

**Comment by Zingam** [comment]
It looks like https://github.com/microsoft/vscode-cmake-tools/issues/532 is implemented, which makes this issue even more relevant. @gcampbell-msft Would you consider triaging this issue now?

**Comment by gcampbell-msft** [comment]
@Zingam Thanks for following up. 

As a workaround, are you able to simply open your project in VS Code as a sudo user?
